### DISCUSSION TITLE
Soft Limit Alarm

### DISCUSSION THREAD

**Discussion by mvackel** [Paciente8159/uCNC#518] [Config and issues] [answered]
When hard and soft limits are enabled, the soft limit alarm seems to be issued only after a few (5 or 6) tries of trespassing it. The motor affected does not move (as expected) but the alarm does not flag.
The following are from UGS console - the soft limit was reached after the first $J command- :

>>> $J=G21G91X-0.1F500
>>> $J=G21G91X-0.1F500
>>> $J=G21G91X-0.1F500
>>> $J=G21G91X-0.1F500
>>> $J=G21G91X-0.1F500
ok
An unexpected error was detected: (ALARM:2) Soft limit alarm. G-code motion target exceeds machine travel. Machine position retained. Alarm may be safely unlocked.
[MSG:Reset to continue]

**Comment by Paciente8159** [comment]
Thanks. I will try to reproduce it

**Comment by Paciente8159** [comment, marked as answer]
I tested and there also seems to be an issue with UGS.
I tried to enable soft limits via UGS firmware settings and even the command line and UGS kept resetting $20 to 0.
Only after running the setup wizard was I able to enable softlimits on UGS.

You are correct. On JOG if you try to send it outside of the soft limits it does not issue an alarm. It simply ignores if the command tries to send the machine to a position outside of it's limits.

I'm not sure if this is a bad thing or not but probably is not similar to Grbl (need to test).

If that is the case I will fix it to send the alarm.

**EDIT**
After a bit more testing the JOG outside limits is working as expected.
Like in Grbl if you try to JOG outside the machine limits you will not get an ALARM but an error message.

In this particular case error 15. UGS simply does not show any warning or notice of this error. But if you do the same on a console or other programs like laserGrbl you will see the just that.

Here is an example with laserGRBL. I was using the future v1.8 branch but the same will happen with current version 1.7.x
![image](https://github.com/Paciente8159/uCNC/assets/12116536/190ac24a-9aa5-424e-af8f-4047f0652e62)
